titus-server-runtime/src/main/java/com/netflix/titus/runtime/endpoint/admission/JobRuntimePredictionSelectors.java

```diff
@@ -0,0 +1,121 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.runtime.endpoint.admission;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Random;
+import java.util.SortedSet;
+
+import com.netflix.archaius.api.annotations.DefaultValue;
+import com.netflix.titus.api.jobmanager.JobAttributes;
+import com.netflix.titus.common.util.CollectionsExt;
+import com.netflix.titus.runtime.connector.prediction.JobRuntimePrediction;
+
+public final class JobRuntimePredictionSelectors {
+
+    private static final JobRuntimePredictionSelector NEVER_SELECTOR = (jobDescriptor, predictions) -> Optional.empty();
+
+    private JobRuntimePredictionSelectors() {
+    }
+
+    public static JobRuntimePredictionSelector never() {
+        return NEVER_SELECTOR;
+    }
+
+    /**
+     * Always select the highest quality prediction.
+     */
+    public static JobRuntimePredictionSelector best() {
+        return best(Collections.emptyMap());
+    }
+
+    /**
+     * Always select the highest quality prediction, with additional metadata.
+     */
+    public static JobRuntimePredictionSelector best(Map<String, String> selectionMetadata) {
+        return (jobDescriptor, predictions) -> CollectionsExt.isNullOrEmpty(predictions.getPredictions())
+                ? Optional.empty()
+                : Optional.of(JobRuntimePredictionSelection.newBuilder().withPrediction(predictions.getPredictions().last()).withMetadata(selectionMetadata).build());
+    }
+
+    /**
+     * Select prediction with the lowest confidence level which is above the provided confidenceThreshold.
+     */
+    public static JobRuntimePredictionSelector aboveThreshold(double runtimeThresholdInSeconds,
+                                                              double sigmaThreshold,
+                                                              Map<String, String> selectionMetadata) {
+        return (jobDescriptor, predictions) -> {
+            SortedSet<JobRuntimePrediction> predictionsSet = predictions.getPredictions();
+            if (CollectionsExt.isNullOrEmpty(predictionsSet) || predictionsSet.size() < 2) {
+                return Optional.empty();
+            }
+
+            JobRuntimePrediction last = predictionsSet.last();
+            if (last.getRuntimeInSeconds() > runtimeThresholdInSeconds) {
+                return Optional.empty();
+            }
+
+            JobRuntimePrediction first = predictionsSet.first();
```

Comment:
verify that the confidence here is `0.05` (and document the assumption), given we assume that when calculating `JobRuntimePredictionUtil.NORM_SIGMA`